### PR TITLE
Add path-related routes

### PR DESCRIPTION
**Pull request by AdvisorChuanChuan**
Create `getCoordinates(fullAddress)` helper function that returns the (longitude, latitude) coordinate of a given address. This is helpful in both station-related and path-related routes, where the server needs to know the exact location of a user-specified address.

There are 3 path-related routes.

- `GET /geocode/:fullAddress`: Simply call the previous helper function.
- `GET /plan/:startLng/:startLat/:destLng/:destLat`: Call OpenRouteService API to obtain the waypoints along the planned path. Here I need to include the API key in our .env file. **Please update yours from our google drive.**
- `POST /routes/nearbyStations`: Utilize the helper function `getWhereClause(filters)` written by @lizyn . Filter stations near the planned path.

### CODE REVIEW

**Review by YichengShen — APPROVED** [review]
LGTM

File: server/routes/paths.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  // Handle the distance constraint along the path
+  const distConstraints = [];
+  for (const coordinate of coordinates) {
+    distConstraints.push(
+      `ST_Distance_Sphere(${stationLocationColName}, ST_SRID(ST_GEOMFROMTEXT('POINT(${
+        coordinate[0]
+      } ${coordinate[1]})'), 4326)) < ${maxDistMile * 1609.34}`
+    );
+  }
```
Comment (lizyn) [inline server/routes/paths.js:78]: I learned a lot from this code snippet.

**Review by lizyn — APPROVED** [review]
looks good to me!

File: server/routes/paths.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  const distConstraintString = distConstraints.join(" OR ");
+
+  // Combine the queries
+  const query = `
+    SELECT DISTINCT sid, ST_Y(${stationLocationColName}) AS longitude, ST_X(${stationLocationColName}) AS latitude
```
Comment (lizyn) [inline server/routes/paths.js:83]: I want to double-check here: does `ST_X` return longitude or latitude?

**Pull request closed without merging**